test/JsonApiDotNetCoreMongoDbExampleTests/IntegrationTests/ReadWrite/Updating/Resources/UpdateResourceTests.cs

```diff
@@ -0,0 +1,502 @@
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using FluentAssertions;
+using JsonApiDotNetCore.Configuration;
+using JsonApiDotNetCore.MongoDb;
+using JsonApiDotNetCore.MongoDb.Repositories;
+using JsonApiDotNetCore.Resources;
+using JsonApiDotNetCore.Serialization.Objects;
+using Microsoft.Extensions.DependencyInjection;
+using MongoDB.Driver;
+using Xunit;
+
+namespace JsonApiDotNetCoreMongoDbExampleTests.IntegrationTests.ReadWrite.Updating.Resources
+{
+    public sealed class UpdateResourceTests
+        : IClassFixture<IntegrationTestContext<TestableStartup>>
+    {
+        private readonly IntegrationTestContext<TestableStartup> _testContext;
+        private readonly WriteFakers _fakers = new WriteFakers();
+
+        public UpdateResourceTests(IntegrationTestContext<TestableStartup> testContext)
+        {
+            _testContext = testContext;
+            
+            _testContext.RegisterResources(builder =>
+            {
+                builder.Add<RgbColor, string>();
+                builder.Add<UserAccount, string>();
+                builder.Add<WorkItem, string>();
+            });
+            
+            _testContext.ConfigureServicesAfterStartup(services =>
+            {
+                services.AddResourceRepository<MongoDbRepository<RgbColor>>();
+                services.AddResourceRepository<MongoDbRepository<UserAccount>>();
+                services.AddResourceRepository<MongoDbRepository<WorkItem>>();
+            });
```

Comment:
Knowing that the example project exclusively uses MongoDB (not a mixed form with EF Core), it is possible to overwrite the built-in open-generic registrations for `EntityFrameworkCoreRepository` with registrations for `MongoDbRepository`, so these calls can be removed.

To do that:
1. In JsonApiDotNetCoreMongoDbExample.Startup, replace the lines with ```services.AddResourceRepository``` with the following:
```c#
services.AddScoped(typeof(IResourceReadRepository<,>), typeof(MongoDbRepository<,>));
services.AddScoped(typeof(IResourceWriteRepository<,>), typeof(MongoDbRepository<,>));
services.AddScoped(typeof(IResourceRepository<,>), typeof(MongoDbRepository<,>));
```
2. Add the same to TestableStartup.ConfigureServices.
3. Likewise, I'd prefer to keep the GettingStarted project as-is.